lib/Fmt_ast.ml

```diff
@@ -261,8 +261,11 @@ let fmt_constant c ?epi {pconst_desc; pconst_loc= loc} =
       str lit $ opt suf char
   | Pconst_char (_, s) -> wrap "'" "'" @@ str s
   | Pconst_string (s, loc', Some delim) ->
-      Cmts.fmt c loc'
-      @@ wrap_k (str ("{" ^ delim ^ "|")) (str ("|" ^ delim ^ "}")) (str s)
+      Cmts.fmt c loc' @@ str (Format_.sprintf "{%s|%s|%s}" delim s delim)
+      (* If a multiline string has newlines in it, it should get treated as a
+         "long" box element. To do so, we append a length-1000 empty
+         string. *)
+      $ fmt_if_k (String.mem s '\n') (str_as 1000 "")
```

Comment:
It's generally not a good idea to put a break at the end of a box as it interferes with the formatting of the containing expression:
```ocaml
let _ =
  {|
    foo
    bar
  |}
  ;
  ""
```

Instead, we write rules that can be followed when formatting the containing expression. For example, `is_simple`, `break_between`, and more defined in `Ast`. This is more or less done in `fmt_args_grouped`.

Reply:
Thanks for pointing this out, I'll take a look at the PR you linked above and try to come up with a fix.

Reply:
It turns out that, as far as I can tell, semicolons after multiline strings still format fine in the janestreet profile.

I have pushed a commit that restricts the new behavior to the janestreet profile, though.